app/models/plan.rb

```diff
@@ -189,8 +189,35 @@ def get_guidance_group_options
     end
     return ggroups.uniq
   end
-
-
+  
+  ##
+  # Sets up the plan for feedback:
+  #  emails confirmation messages to owners
+  #  emails org admins and org contact 
+  #  adds org admins to plan with the 'reviewer' Role
+  def request_feedback(user)
+    # Share the plan with each org admin as the reviewer role
+    admins = user.org.org_admins
+    admins.each do |admin|
+      role = Role.create!(plan: self, user: admin, access: 16)
+    end 
+    
+    # Send an email confirmation to the owners and co-owners
+    self.owner_and_coowners.each do |owner|
+      UserMailer.feedback_confirmation(owner, self, user).deliver_now
+    end
+  
+    # Send an email to all of the org admins as well as the Org's administrator email
+    if user.org.contact_email.present?
+      admins << User.new(email: user.org.contact_email, firstname: user.org.contact_name)
+    end
+    admins.each do |admin|
+      UserMailer.feedback_notification(admin, self, user).deliver_now
+    end
+    
+    self.feedback_requested = true
+    self.save!
```

Comment:
this is safer right before sending any email. In case it failed, both creators and potential reviewers will receive an email for a non-existing request feedback for a plan.

Reply:
Ok. This is a better solution. I refactored to add the Roles to `self.roles` and then send the emails if `self.save!` succeeds. The flag and roles are saved in a single transaction.